src/components/MdnRequest.vue

```diff
@@ -0,0 +1,47 @@
+<template>
+	<div v-if="hasMdnRequest" id="mail-message-has-mdn-request">
+		{{
+			t('mail', 'The sender of this message has asked to be notified when you read this message.')
+		}}
+		<button @click="sendMdn">
+			{{ t('mail', 'Notify the sender') }}
+		</button>
+	</div>
+</template>
+
+<script>
+import logger from '../logger'
+import { sendMdn } from '../service/MessageService'
+
+export default {
+	name: 'MdnRequest',
+	props: {
+		message: {
+			required: true,
+			type: Object,
+		},
+	},
+	data() {
+		return {
+			hasMdnRequest: false,
+		}
+	},
+	mounted() {
+		this.hasMdnRequest = !this.message.flags.mdnsent && this.message.dispositionNotificationTo && this.message.dispositionNotificationTo.length > 0
+	},
+	methods: {
+		async sendMdn() {
+			logger.debug('send return receipt')
+
+			await sendMdn(this.message.databaseId)
```

Comment:
wrap with try-catch and use `showError` from `@nextcloud/dialogs` to show a toast on error